frontend/packages/dev-console/src/components/project-access/ProjectAccess.tsx

```diff
@@ -75,27 +81,24 @@ const ProjectAccess: React.FC<ProjectAccessProps> = ({
         (o1) => !updateRoles.find((o2) => o1.roleBindingName === o2.roleBindingName),
       );
     }
-
+    updateRoles.push(...updateRolesWithMultipleSubjects);
     const roleBindingRequests = [];
     roleBinding.metadata.namespace = namespace;
 
-    removeRoles = _.filter(removeRoles, (removeRole) => {
-      const groupedRole = getGroupedRole(removeRole, roleBindings.data);
-      if (groupedRole) {
-        roleBindingRequests.push(sendK8sRequest(Verb.Patch, groupedRole));
-        return false;
-      }
-      return true;
-    });
-
     if (!_.isEmpty(updateRoles)) {
-      roleBindingRequests.push(...sendRoleBindingRequest(Verb.Patch, updateRoles, roleBinding));
+      roleBindingRequests.push(
+        ...sendRoleBindingRequest(Verb.Patch, updateRoles, _.clone(roleBinding)),
+      );
     }
     if (!_.isEmpty(removeRoles)) {
-      roleBindingRequests.push(...sendRoleBindingRequest(Verb.Remove, removeRoles, roleBinding));
+      roleBindingRequests.push(
+        ...sendRoleBindingRequest(Verb.Remove, removeRoles, _.clone(roleBinding)),
+      );
     }
     if (!_.isEmpty(newRoles)) {
-      roleBindingRequests.push(...sendRoleBindingRequest(Verb.Create, newRoles, roleBinding));
+      roleBindingRequests.push(
+        ...sendRoleBindingRequest(Verb.Create, newRoles, _.clone(roleBinding)),
+      );
```

Comment:
Small suggestion here: The caller should not know that `sendRoleBindingRequest` changes the roleBinding parameter. This should be an implementation detail.

When its easier to mutate the roleBinding parameter instead of using a spread operator in `sendRoleBindingRequest` you can also clone it there.